test/functional/provider/perl_spec.lua

```diff
@@ -0,0 +1,42 @@
+local helpers = require('test.functional.helpers')(after_each)
+local eq, clear = helpers.eq, helpers.clear
+local missing_provider = helpers.missing_provider
+local command = helpers.command
+local write_file = helpers.write_file
+local eval = helpers.eval
+local retry = helpers.retry
+
+do
+  clear()
+  local reason = missing_provider('perl')
+  if reason then
+    pending(string.format("Missing perl host, or perl version is too old (%s)", reason), function() end)
+    return
+  end
+end
+
+before_each(function()
+  clear()
+end)
+
+describe('perl host', function()
+  teardown(function ()
+    os.remove('Xtest-perl-hello.pl')
+  end)
+
+  it('works', function()
+    local fname = 'Xtest-perl-hello.pl'
+    write_file(fname, [[
+      use v5.22.1;
+      use Neovim::Ext;
+      use Neovim::Ext::MsgPack::RPC;
+
+      my $session = Neovim::Ext::MsgPack::RPC::socket_session($ENV{NVIM_LISTEN_ADDRESS});
+      my $nvim = Neovim::Ext::from_session($session);
+      $nvim->command('let g:job_out = "hello"');
```

Comment:
@jacquesg What's the simplest plugin abstraction that does this within one file (no `rplugin/perl/`)? I can't figure out how to use the Plugin class for this without creating a new file. For reference, I want to do the equivalent of what's in the nodejs_spec.lua.

Reply:
Nevermind. I found a simple workaround.